service-control/src/main/java/fi/nls/oskari/control/ActionParameters.java

```diff
@@ -285,4 +292,19 @@ public String getAPIkey() {
         // TODO: use something better than session id
         return getRequest().getSession().getId();
     }
+
+    /**
+     * Get play load JSON
+     * @return
+     */
+    public JSONObject getPayLoadJSON() {
+        JSONObject payloadJSON = null;
+        HttpServletRequest req = this.getRequest();
+        try (InputStream in = req.getInputStream()) {
+            final byte[] json = IOHelper.readBytes(in);
+            final String jsonString = new String(json, StandardCharsets.UTF_8);
+            return new JSONObject(jsonString);
+        } catch (Exception ignored) {}
+        return payloadJSON;
```

Comment:
Instead of silently swallowing the exception this method should probably throw it so that the caller can decide how he/she wants to handle it.